### PR TITLE
ban usage of `require.Equal` when testing for errors

### PR DESCRIPTION
**Pull request by dhrubabasu**

### CODE REVIEW

File: scripts/lint.sh [hunk 1]
```diff
@@ -125,6 +125,15 @@ function test_require_no_error_inline_func {
   fi
 }
 
+function test_require_equal_error {
+  if ggrep -R -o -P 'require.Equal.+?err(\)|,)' .; then
```
Comment (dhrubabasu) [inline scripts/lint.sh:129]:
```suggestion
  if grep -R -o -P 'require.Equal.+?err(\)|,)' .; then
```

File: database/test_database.go [hunk 2]
```diff
@@ -95,7 +95,7 @@ func TestSimpleKeyValue(t *testing.T, db Database) {
 	require.False(has)
 
 	_, err = db.Get(key)
-	require.Equal(ErrNotFound, err)
+	require.ErrorIs(err, ErrNotFound)
```
Comment (dhrubabasu) [inline database/test_database.go:98]: this doesn't fail the linter (and I don't think there's a regex we can add to consistently test for this). Just need to be more vigilant going forward

**Review by danlaine — APPROVED** [review]

**Review by gyuho — APPROVED** [review]
🥇

File: database/test_database.go [hunk 3]
```diff
@@ -173,14 +173,19 @@ func TestSimpleKeyValueClosed(t *testing.T, db Database) {
...
 	_, err = db.Has(key)
-	require.Equal(ErrClosed, err)
+	require.ErrorIs(err, ErrClosed)
 
 	_, err = db.Get(key)
-	require.Equal(ErrClosed, err)
+	require.ErrorIs(err, ErrClosed)
+
+	err = db.Put(key, value)
+	require.ErrorIs(err, ErrClosed)
```
Comment (abi87) [inline database/test_database.go:182]: isn't there another PR where we inline this stuff? If so, what's the policy to inline/not inline?

Comment (dhrubabasu) [reply database/test_database.go:182]: We don't inline functions for `ErrorIs`. We do for `NoError`

**Comment by dhrubabasu** [comment]
Closing for now, we should revisit after enabling `goerr113` as per @StephenButtolph's earlier comment.

**Pull request closed without merging**